### ISSUE TITLE
Soil water availability 

### ISSUE THREAD

**Issue by JohannesOberpriller** [trotsiuk/r3PG#51]
Hi Vova, 

I was wondering what maximum and minimum available soil water (asw_min, asw_max) actually is and about the diference to field capacity and permanent wilting point? I was also wondering if it is only the difference between maximum and minimum that matters in the end as water storage in the model ?

Best wishes,
Johannes

**Comment by trotsiuk** [comment]
Hi @JohannesOberpriller 

According to Landsberg and Waring (1997) it is stated in the paragraph 2.2 `soil water contents = maximum available waters in the rooting zone`.

And I believe it is not only the difference between maximum and minimum. The `asw_max` is used to calculate the [soil water modifier](https://github.com/trotsiuk/r3PG/blob/f7c626044eca5a088b1452cd58711ab06dc742c1/pkg/src/md_3PG.f95#L333) and [soil conductance](https://github.com/trotsiuk/r3PG/blob/f7c626044eca5a088b1452cd58711ab06dc742c1/pkg/src/md_3PG.f95#L375).

And the available soil water can't be lower than `asw_min`.

But maybe @DavidForrester can elaborate more on it.


Reference
Landsberg, J. J., & Waring, R. H. (1997). A generalised model of forest productivity using simplified concepts of radiation-use efficiency, carbon balance and partitioning. Forest Ecology and Management, 95(3), 209–228. https://doi.org/10.1016/S0378-1127(97)00026-1

**Comment by florianhartig** [comment]
Hi Vova,

the reason for Johannes' question was also how to determine these values from empirical data. Did you have these values in your data for Switzerland, or did you somehow derive them from field capacity or something like that?

Cheers,
F

**Comment by GakenouOF** [comment]
I am also interested in this issue particularly the `asw_min`. Some author estimated `asw_max` as asw (derived from FC - WP) x rooting depth. But no clear definition of what `asw_min` is. Default was set as zero.

**Comment by DavidForrester** [comment]
Thanks for the question. The asw_max is pretty much the same as the field capacity, and quantifies the maximum amount of water the soil type can hold. The permanent wilting point is when the pressure required to extract water from the soil is too high for the plants (often about 1500kPa) but it varies a bit between species. So the asw_min = 0 would be representing the permanent wilting point. The asw_min is therefore often set to 0 in 3-PG, but it occasionally it can also be reasonable to set it to > 0, e.g. when the user knows that the plants have access to a deep permanent water source, or when there is irrigation etc. In these situations, it needs to be calculated based on information about how much water is being applied (irrigation) or how much the plants can potentially access (e.g. from a deep water table) after the precipitation water has been used.

To derive asw_max, the equation mentioned sounds reasonable: (derived from FC - WP) x rooting depth.

**Comment by trotsiuk** [comment]
For Switzerland we have been using the old data from the Frei et al. 1980.
But for Europe, I experimented once using the [European Soil Database Derived data ](https://esdac.jrc.ec.europa.eu/content/european-soil-database-derived-data).

Frei, E., Voegt, U., Flueckiger, R., Brunner, H., Schai, F., & Haeberli, R. (1980). Bodeneignungskarte der Schweiz. Neuchâtel, Switzerland: Bundesamt für Statistik, Sektion Geoinformation.

**Comment by florianhartig** [comment]
Thanks @DavidForrester. Vova, but how did you derive the min / max values from those data sources? I assume they are not provided directly. Did you use the idea David is sketching?

**Comment by trotsiuk** [comment]
In Frei et al. 1980 they were provided, but how accurate are the estimation is unclear.
For ESDD yes, the way David described

**Comment by JohannesOberpriller** [comment]
Thanks @DavidForrester and @trotsiuk for the clarification. We will use @DavidForrester sketch to calculate asw_max.

**Comment by florianhartig** [comment]
Reopening this as a note to self (or others if you want to) to add some more documentation on this point!

**Comment by twest820** [comment]
Cool to see this reopened. It seems to me all of the current 3-PG documentation assumes users know to interpret asw_max and asw_min as @DavidForrester describes here. And that users know asw_max - asw_min = PAWC (plant available water capacity, also referred to as plant available water or available water capacity and abbreviated PAW or AWC). And that users know to estimate PAWC as a hydraulic soil property from a PTF (pedotransfer function) appropriate for their sites and tree rooting depths. Or some equivalent thereof. For example, the soil databases for all of my study sites give approximate textures and horizon thicknesses but about half of the sites don't have PAWC information. The other half have a PAWC point value that's an agricultural estimate (it's probably n = 1 from a type pedon) and therefore likely an underestimate for large, perennial species such as trees.

So I believe even minimal guidance to 3-PG users in this regard would be helpful. Some papers I've found informative for general context on PTFs are [Patil and Singh 2016](https://doi.org/10.1016/S1002-0160(15)60054-6), [Dai et al. 2019](https://doi.org/10.1029/2019MS001784), and [Zhang et al. 2019](https://doi.org/10.2136/vzj2018.08.0151). In R, [soilDB](https://rdocumentation.org/packages/soilDB/versions/2.6.2/topics/ROSETTA) provides an interface to Rosetta3 but the documentation doesn't explain how to calculate PAWG by integrating Mualem-van Genuchten through the rooting depth (log₁₀ α and log₁₀ n are returned and Ψ is in cm, so it's not entirely plug and chug). I think most other PTFs are outside of R. There is perhaps something of a feature gap here in that I don't know of PTFs which readily yield a PAWC distribution and I'm not sure if there's a way to tell BayesianTools of a joint distribution for `asw_max` and `asw_min`. For the soils I've looked at `asw_max` is about eight times larger than `asw_min` so, at least for those, lack of a joint distribution probably isn't a major issue.

A related detail I've noticed is Table 2 of the 3-PGpjs 2.7 manual indicates only integer values for `soil_class` and implies all other values trigger use of `SWconst` and `SWpower`. However, r3PG ([md_3PG.f95:95-107](https://github.com/trotsiuk/r3PG/blob/master/pkg/src/md_3PG.f95#L95) at the moment) uses negative soil classes for this and treats a positive `soil_class` as a real value yielding continuous values of `SWconst` and `SWpower`. I find this preferable to strictly following the 3-PGpjs spec but it'd probably be good if the docs for `d_site` and `d_species` noted the interactions. I found this the hard way when I noticed soil water manipulations weren't giving expected results.

**Comment by trotsiuk** [comment]
Not sure Tom what do you mean with
>However, r3PG (md_3PG.f95:95-107 at the moment) uses negative soil classes for this and treats a positive soil_class as a real value yielding continuous values of SWconst and SWpower

But I would say `r3PG` have is identical implementation of the original `3-PGpjs 2.7` or `3-PGmix` for the `soil class`

http://people.forestry.oregonstate.edu/richard-waring/forest-growth-model-3-pg
```
Assign the SWconst and SWpower parameters for this soil class
  If soilClass > 0 Then
    'standard soil type
    SWconst = 0.8 - 0.1 * soilClass
    SWpower = 11 - 2 * soilClass
  ElseIf soilClass < 0 Then
    'use supplied parameters
    SWconst = SWconst0
    SWpower = SWpower0
  Else
    'no soil-water effects
    SWconst = 999
    SWpower = SWpower0
  End If
```

**Comment by trotsiuk** [comment]
>  It seems to me all of the current 3-PG documentation assumes users know to interpret asw_max and asw_min

It might be worth to address this point at the main central place for most of 3-PG implementations 

https://3pg.forestry.ubc.ca

https://3pg.forestry.ubc.ca/software/

**Comment by DavidForrester** [comment]
Thanks for the tip. Some information about calculating available soil water is now available in the manual, which is available from https://sites.google.com/site/davidforresterssite/home/projects/3PGmix

**Comment by twest820** [comment]
Cool. What I get from [https://sites.google.com/site/davidforresterssite/home/projects/3PGmix](https://sites.google.com/site/davidforresterssite/home/projects/3PGmix) is a set of intro paragraphs with 14 references (sorted 14→1 rather than 1→14) and no mention of soil water calculations. [https://sites.google.com/site/davidforresterssite/home/projects/3PGmix/3pgmixdownload](https://sites.google.com/site/davidforresterssite/home/projects/3PGmix/3pgmixdownload) just gives an error ("accounts.google.com refused to connect").

So I think this was an update to 3PGmix_UserManual.pdf in [3PGmix.zip](https://drive.google.com/file/d/1SSeiDs8XWkPAgwuRyKGTxb8-nRAIzyHO/view), presumably at the end of section 10.3 on page 47. A caution is that, while it's common to assume coarse fragments (soil particles 2+ mm) hold no water, the few studies which have looked at this seem to consistently find the accuracy of this assumption is questionable. My sense of the literature is it's not well understood what fraction of water in coarse fragments is plant available but most authors seem to think it can be significant to overall PAWC, at least in some soils. [Robertson et al. 2016](https://doi.org/10.1016/j.geoderma.2020.114912) is one of the more recent papers on the topic. The other caution I'd make is the current text doesn't mention stopping the integration at rooting depth, whether at the limit of a species' soil structure and climate adaptive response or at a horizon whose bulk density obstructs roots. For example, I'm setting up a 3-PG model at the moment for a tree species which is often observed to root to a 1-1.2 m depth on a sandy site with a very gravelly layer at about 43 cm whose documentation notes it blocks roots. So uncertainty in PAWC there is more from variation in the gravel depth and silt and clay content than it is from uncertainty about roots reaching a given depth.

> But I would say `r3PG` have identical implementation of the original `3-PGpjs 2.7` or `3-PGmix` for the `soil class`

The thing to do then would be to correct the table in the 3-PGpjs user manual. That might be out of scope for this repo, though.

**Comment by Rasilgon** [comment]
Hi @florianhartig  and @trotsiuk,
We checked time ago how to derive asw_max from the ESDD, and I'm not sure if understand how @trotsiuk  used the data for this.  There is already a variable "Total available soil water content" (TAWC), calculate as:
![image](https://user-images.githubusercontent.com/10464055/160599715-6934136b-2656-4e16-9774-5524fcf79eb7.png)

Which is available for the topsoil (0-30 cm, max d=30 cm) and subsoil (30-100 cm, max d=70).  TAWC would seem to be more or less equivalent to asw_max., right? 
From what I understand, you did not use this one, but the formula (FC - WP) x rooting depth, and found FC and WP in ESDD, which we did not find. Rooting depth is available and given classes of 20 cm up to > 80 cm, which fits the ranges of the subsoil. So far, I could find in the ESDD reports  whether rooting depth and max subsoil depth refer to same depth value, (sorry, I am not expert in soil stuff), but it does seem so.  Anyway, if so,  asw_max could just be derived as `TAWC topsoil + TAWC subsoil`,

**Comment by JohannesOberpriller** [comment]
Hi,

I am out of sience. This email will not be read or forwarded. 

Best
Johannes Oberpriller 

>>> Trotsiuk Volodymyr ***@***.***> 4.7.23 09:43 >>>

Closed #51 as completed.

-- 
Reply to this email directly or view it on GitHub:
https://github.com/trotsiuk/r3PG/issues/51#event-9718254611
